src/packet.rs

```diff
@@ -610,18 +573,9 @@ pub fn negotiate_version(
     b.put_u8(first)?;
     b.put_u32(0)?;
 
-    // Invert client's scid and dcid.
-    let mut cil: u8 = 0;
-    if !scid.is_empty() {
-        cil |= ((scid.len() - 3) as u8) << 4;
-    }
-
-    if !dcid.is_empty() {
-        cil |= ((dcid.len() - 3) as u8) & 0xf;
-    }
-
-    b.put_u8(cil)?;
+    b.put_u8(scid.len() as u8)?;
```

Comment:
We weren't doing this before, but it would probably be a good idea that the application doesn't pass too long connection IDs.

Reply:
This is still valid, but we can do this in a seaprate PR.

Reply:
oops I missed this one, agree can do in a separate PR.